In [24]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score
from sklearn.metrics import confusion_matrix


In [3]:
Data = pd.read_csv("train.csv")

### 1. View the first 5 rows using .head()

In [4]:
Data.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


### 2. Get dataset information using .info()

In [5]:
Data.info()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 83.7 KB


### 3. Get statistical summary using .describe()

In [6]:
Data.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


### 4. Number of rows and columns

In [7]:
print("Rows and Columns:", Data.shape)
print("Number of Rows:", Data.shape[0])
print("Number of Columns:", Data.shape[1])

Rows and Columns: (891, 12)
Number of Rows: 891
Number of Columns: 12


### 5. Columns with missing values

In [8]:
missing_values = Data.isnull().sum()
print(missing_values[missing_values > 0])

Age         177
Cabin       687
Embarked      2
dtype: int64


### 6. Identify categorical and numerical columns

In [9]:
# Numerical Columns
numerical_columns = Data.select_dtypes(include=['int64', 'float64']).columns
print("Numerical Columns:")
print(list(numerical_columns))

Numerical Columns:
['PassengerId', 'Survived', 'Pclass', 'Age', 'SibSp', 'Parch', 'Fare']


In [10]:
# Categorical columns
categorical_columns = Data.select_dtypes(include=['object']).columns
print("Categorical Columns:")
print(list(categorical_columns))

Categorical Columns:
['Name', 'Sex', 'Ticket', 'Cabin', 'Embarked']


C:\Users\laptop villa\AppData\Local\Temp\ipykernel_5312\1010423723.py:2: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  categorical_columns = Data.select_dtypes(include=['object']).columns


####
Data Story

The Titanic training dataset contains **891 rows** and **12 columns**, providing information about passengers and whether they survived. Initial exploration shows that the **Age**, **Cabin**, and **Embarked** columns contain missing values, with **Cabin** having the largest number of missing entries. The dataset includes both **numerical** features (such as Age, Fare, and SibSp) and **categorical** features (such as Sex, Embarked, and Cabin). This suggests that data cleaning and preprocessing will be necessary before building a machine learning model. Understanding these characteristics is an important first step toward predicting passenger survival.